### ISSUE TITLE
`__ne__` does not defer to custom `__eq__`

### ISSUE THREAD

**Issue by Solumin** [jcrist/msgspec#585]
### Description

I've encountered unexpected behavior with custom-defined `__eq__` methods:

```py
class A(msgspec.Struct, eq=False):
  x: int
  y: int
  def __eq__(self, other):
    return self.__class__ == other.__class__ and self.x == other.x

x = A(1, 2)
y = A(1, 3)
print(x == y)  # True
print(x != y)  # True  <-- unexpected!
```
The [default implementation][1] of `__ne__` just defers to `__eq__`, so `x != y` should be `False`. It seems that `Struct` has its own `__ne__` implementation that does not do this.

Admittedly, this is a really weird case that would probably be solved by making `y` a private field (#199). I'd also be happy with just updating the docs to state that if you provide a custom `__eq__` that you should also define `__ne__`.

[1]: https://docs.python.org/3/reference/datamodel.html#:~:text=For%20__ne__()%2C%20by%20default%20it%20delegates%20to%20__eq__()%20and%20inverts%20the%20result%20unless%20it%20is%20NotImplemented.

**Comment by jcrist** [comment]
Thanks for the excellent reproducible issue, this has been fixed in #593.

> Admittedly, this is a really weird case that would probably be solved by making y a private field (https://github.com/jcrist/msgspec/issues/199).

As part of that work I also intend to support field level configuration for `eq`/`repr`/`init`, matching the behavior of [dataclasses](https://docs.python.org/3/library/dataclasses.html#dataclasses.field) and [attrs](https://www.attrs.org/en/stable/api.html#attrs.field). When that's done, you should be able to do something like this to get what you want:

```python
class A(msgspec.Struct):
  x: int
  y: int = msgspec.field(eq=False)
```

**Comment by Solumin** [comment]
Thank you for fixing this so quickly!

> As part of that work I also intend to support field level configuration for eq/repr/init, matching the behavior of [dataclasses](https://docs.python.org/3/library/dataclasses.html#dataclasses.field) and [attrs](https://www.attrs.org/en/stable/api.html#attrs.field).

Excellent, I think that will be a very helpful feature.
